broker.go

```diff
@@ -768,24 +782,27 @@ func (b *Broker) BatchConsumer(conf ConsumerConf) (BatchConsumer, error) {
 }
 
 func (b *Broker) consumer(conf ConsumerConf) (*consumer, error) {
-	offset := conf.StartOffset
-	if offset < 0 {
-		switch offset {
-		case StartOffsetNewest:
-			off, err := b.OffsetLatest(conf.Topic, conf.Partition)
-			if err != nil {
-				return nil, err
-			}
-			offset = off
-		case StartOffsetOldest:
-			off, err := b.OffsetEarliest(conf.Topic, conf.Partition)
-			if err != nil {
-				return nil, err
-			}
-			offset = off
-		default:
-			return nil, fmt.Errorf("invalid start offset: %d", conf.StartOffset)
+	var offset int64
+	switch {
+	case conf.StartFrom == StartFromAbsolute:
+		offset = int64(conf.StartOffset)
+	case conf.StartOffset != 0:
+		return nil, fmt.Errorf("Set AbsoluteStartOffset uint64 (%d) with non-StartOffsetAbsolute (%d).",
+			conf.StartOffset, conf.StartFrom)
+	case conf.StartFrom == StartFromNewest:
+		off, err := b.OffsetLatest(conf.Topic, conf.Partition)
+		if err != nil {
+			return nil, err
+		}
+		offset = off
+	case conf.StartFrom == StartFromOldest:
+		off, err := b.OffsetEarliest(conf.Topic, conf.Partition)
+		if err != nil {
+			return nil, err
 		}
+		offset = off
+	default:
+		return nil, fmt.Errorf("invalid start offset: %d", conf.StartFrom)
```

Comment:
now that I look at it I think this error message could be more precise, since it's not the offset that's invalid but the StartFrom.  maybe "invalid choice for StartFrom" would be better?